src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -345,6 +346,58 @@ SIMPLE_MATCHER_PASS_DEFINITION(EliminateGather,
                                ov::op::v7::Gather,
                                ov::op::v8::Gather);
 
+pass::EliminateReduceReshape::EliminateReduceReshape() {
+    MATCHER_SCOPE(EliminateReduceReshape);
+    using namespace pass::pattern;
+    auto axes = wrap_type<ov::op::v0::Constant>();
+    auto reduce_pattern = wrap_type<ov::op::util::ArithmeticReductionKeepDims>({any_input(), axes});
+    auto requested_shape_pattern = wrap_type<ov::op::v0::Constant>();
+    auto reshape_pattern =
+        wrap_type<ov::op::v1::Reshape>({reduce_pattern, requested_shape_pattern}, consumers_count(1));
+
+    matcher_pass_callback callback = [=](Matcher& m) {
+        auto pattern_map = m.get_pattern_map();
+        auto reshape_node = m.get_match_root();
+        auto reduce_node = pattern_map.at(reduce_pattern);
+
+        auto reshape = ov::as_type_ptr<ov::op::v1::Reshape>(reshape_node);
+        auto reduce = ov::as_type_ptr<ov::op::util::ArithmeticReductionKeepDims>(reduce_node);
+        if (!reshape || !reduce) {
```

Comment:
Could it be here? Pattern defines particular types of nodes. Should we check it here?

Reply:
Yes, we need this cast because I use the specific method for Reduce in the code below

Reply:
I am not about cast, I am about check if not nullptr

Reply:
these are typical checks in c++ code after dynamic cast